src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/BaseTest.cs

```diff
@@ -55,6 +49,20 @@ static BaseTest ()
 			}
 		}
 
+		protected static string RunAdbCommand (string command, bool ignoreErrors = true)
+		{
+			string ext = Environment.OSVersion.Platform != PlatformID.Unix ? ".exe" : "";
+			var home = Environment.GetFolderPath (Environment.SpecialFolder.Personal);
+			string adb = Path.Combine (Environment.GetEnvironmentVariable ("ANDROID_SDK_PATH") ?? $"{home}/android-toolchain/sdk", "platform-tools", "adb" + ext);
```

Comment:
This should use `string.IsNullOrEmpty()`, not `??`, as -- at least in bash -- there is no way to "unset" an environment variable, you can only clear it's contents:

```sh
$ export ANDROID_SDK_PATH=/whatever
$ export ANDROID_SDK_PATH=
# Now $ANDROID_SDK_PATH is ''
```

In such an environment, `Environment.GetEnvironmentVariable ("ANDROID_SDK_PATH")` will return `""`, not `null`.